utils_cv/detection/model.py

```diff
@@ -167,109 +185,112 @@ def fit(self, epochs: int, print_freq: int = 10) -> None:
         return self.results
 
-    def get_model(self) -> nn.Module:
-        """ returns the model object. """
-        return self.model
-
-    def inference(self, im: np.ndarray) -> Dict[Any, Any]:
-        """ Performs inferencing on an image path.
+    def predict(
+        self,
+        im_or_path: Union[np.ndarray, Union[str, Path]],
+        threshold: int = 0.6,
+    ) -> List[AnnotationBbox]:
+        """ Performs inferencing on an image path or image.
 
         Args:
-            im: the image array which you can get from `Image.open(path)`
+            im_or_path: the image array which you can get from `Image.open(path)` OR a
+            image path
+            threshold: the threshold to use to calculate whether the object was
+            detected
 
         Raises:
             TypeError is the im object is a path or str to the image instead of
             an nd.array
 
-        Return the prediction dictionary object
+        Return a list of AnnotationBbox
         """
-        if isinstance(im, (str, Path)):
-            raise TypeError("Pass in a np.ndarray, not image path.")
+        im = (
+            Image.open(im_or_path)
+            if isinstance(im_or_path, (str, Path))
+            else im_or_path
+        )
 
         transform = transforms.Compose([transforms.ToTensor()])
         im = transform(im).cuda()
-        model = self.get_model().eval()  # eval mode
+        model = self.model.eval()  # eval mode
         with torch.no_grad():
             pred = model([im])
-        return pred
+        labels = self.dataset.labels
+        return get_bounding_boxes(pred, labels=labels, threshold=threshold)
 
-    def show_detection_vs_ground_truth(self, idx: int, ax: plt.axes) -> None:
-        """ Plots the bounding box predictions and ground truth
+    def predict_dl(
+        self, dl: DataLoader, threshold: int = 0.6
+    ) -> List[AnnotationBbox]:
+        """ Predict all images in a dataloader object.
 
         Args:
-            idx: the index of the dataset to visualize
-            ax: axes to draw on
+            dl: the dataloader to predict on
+            threshold: iou threshold for a positive detection
 
-        Returns nothing but plots graph.
+        Returns a list of AnnotationBboxes
         """
-        im_path = (
-            self.dataset.root
-            / self.dataset.image_folder
-            / self.dataset.ims[idx]
-        )
-        im = Image.open(str(im_path))
-
-        # plot prediction boxes
-        pred = self.inference(im)
-        pred_labels, pred_boxes = get_bounding_boxes(pred)
-        pred_params = PlotSettings(rect_color=(255, 0, 0), text_size=0)
-        im = plot_boxes(
-            im,
-            pred_boxes,
-            [self.dataset.categories[l] for l in pred_labels],
-            plot_settings=pred_params,
-        )
+        pred_generator = self.predict_batch(dl, threshold=threshold)
+        detections = [pred for preds in pred_generator for pred in preds]
```

Comment:
Maybe rename "detections" to "bboxes" since that is more clear. Or a different name as long as its consistent in the code (e.g. predictions vs. detections: think I have seen both used)

Reply:
do you have a suggestion for the long syntax? afaik it is the shortest/cleanest way to do a nested list comprehension.

Reply:
I am likely wrong: would something like this here also work? 
     detections =list(pred_generator)